src/array.jl

```diff
@@ -212,6 +212,31 @@ end
 
             convert($A{T, N, R}, A)
         end
+
+        function Base.vcat{T,N,R}(A1::$A{T, N, R}, An::$A...)
+            As = (A1, An...)
+            levels = unique(T[[a.pool.levels for a in As]...;])
```

Comment:
Warning, the `refs` field codes refer to the `index`, not to the `levels` (which can be in a different order). Also, use high-level methods: `index(a.pool)` instead of `a.pool.index` (here and below).